src/Diagnostics.DataProviders/AscClient.cs

```diff
@@ -0,0 +1,342 @@
+            }
+
+            var response = await httpClient.SendAsync(request, cancellationToken);
+            logger.LogDataProviderMessage(requestId, "AscDataProvider", $"url:{request.RequestUri}, statusCode:{response.StatusCode}");
+            return response;
+        }
+
+        /// <summary>
+        /// Generates a JSON string containing request and response headers along with the response content. Is used for logging purposes.
+        /// </summary>
+        /// <param name="response">HTTP Response message.</param>
+        /// <returns>JSON formatted header and body content.</returns>
+        private async Task<string> GetRequestDetailsForLogging(HttpResponseMessage response)
+        {
+            HTTPResponseLog logMessage = new HTTPResponseLog();
+            if (await logMessage.Initialize(response))
+            {
+                return JsonConvert.SerializeObject(logMessage);
+            }
+            else
+            {
+                return string.Empty;
+            }
+        }
+
+        /// <summary>
+        /// Class to represent HTTP Request / Response. Used for logging purposes.
+        /// </summary>
+        public class HTTPResponseLog
+        {
+            /// <summary>
+            /// URI for the HTTP request.
+            /// </summary>
+            public string requestUri;
+
+            /// <summary>
+            /// HTTP method. GET / POST etc.
+            /// </summary>
+            public HttpMethod method;
+
+            /// <summary>
+            /// Request header Collection.
+            /// </summary>
+            public HttpRequestHeaders requestHeaders;
+
+            /// <summary>
+            /// Request body. Empty in case of a GET request.
+            /// </summary>
+            public string requestBody;
+
+            /// <summary>
+            /// Response status code.
+            /// </summary>
+            public HttpStatusCode responseStatusCode;
+
+            /// <summary>
+            /// Response header Collection.
+            /// </summary>
+            public HttpResponseHeaders responseHeaders;
+
+            /// <summary>
+            /// Response body.
+            /// </summary>
+            public string responseBody;
+
+            /// <summary>
+            /// Initializes the <see cref="HTTPResponseLog"/> class.
+            /// </summary>
+            /// <param name="response">HTTP Response object that you need to create a log for.</param>
+            /// <returns>True if initialized and false otherwise.</returns>
+            public async Task<bool> Initialize(HttpResponseMessage response)
+            {
+                try
+                {
+                    requestUri = response.RequestMessage.RequestUri.ToString();
+                    method = response.RequestMessage.Method;
+                    response.RequestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", "WW91IHdpc2ggSSdkIGxvZyB0aGF0Lg==");
```

Comment:
Need to remove the secret from the code.
Here is how Antares manage secrets and consume those:
- Add a keyvault entry for this secret
- Add a new definition in GeoRegionProd.Definitions and supply the value using keyvault.
  Something like:  <redefine name="ASCSecretName" value="@(Antares/Diagnostics/ASCSecretValue)" />
- Then consume it here.

Talk to Sergey or Andrey. They can help you add this value to Keyvault.

Reply:
I was just having fun there :) Secrets are coming from the config.

It's base64 for "You wish I'd log that."

Reply:
oh haha... you got me there :)
Lets remove that... so it doesnt cause any further questions/confusions by anybody... lol

Reply:
I mean seriously...You are that creative :P